R/edgeThresh.R

```diff
@@ -0,0 +1,461 @@
+# edgeThresh.R
+
+#' Define edge-removal threshold parameter δ for each target subnet-order, from
+#' random permutations of the input EGG object
+#'
+#' \code{THRESH} creates N permutations of the input graph EGG by randomly
+#' choosing 2 edges and swapping their target endpoints (i.e. B in the
+#' edge A → B). Finds the value δ (edge weight) for each random graph such that
+#' the maximum size of any component in the graph is Lmax, and returns the
+#' median of N δ values.
+#'
+#' @section Swapping Edges:
+#'     Bidirectional edges must only be swapped with each other, Since all 4
+#'     incident nodes in 2 bidirectional edges are receiving endpoints, one node
+#'     will be randomly selected from each edge, and a new edge is formed
+#'     between the selected as well as the remaining vertices.
+#'     At each unidirectional edge swap, only the receiving endpoints are
+#'     swapped e.g. (A → B, C → D) to (A → D, C → B).
+#'     In both cases, a swap should NOT result in:
+#'
+#'     "Multiple edges": Possibly result of multiple bidirectional edge swaps:
+#'             A                             A
+#'          //   \\              →                 //
+#'         B   →   C                     B  ↔ →  C
+#'
+#'     OR:
+#'     Where an edge belonging to vertex (A) which already has an outgoing edge to
+#'     another vertex (B) is switched with another incoming edge to vertex B,
+#'     resulting in (A) having two outgoing edges to (B).
+#'
+#'        ( )     ( )                  ( )  ←  ( )
+#'         ↑       ↓             →
+#'        (A)  →  (B)                  (A)   ⇉  (B)
+#'
+#'     "Self-edges": e.g. (A ↔ B, C ↔ A) to (A ↔ A, B ↔ C)
+#'     Where in the case of:
+#'     'Unidirectional edges': Precedent of one edge is the antecedent of the
+#'     other edge to be swapped.
+#'     'Bidirectional edges': The edge pair to be swapped involves less than 4
+#'     distinct nodes.
+#'     "Disconnected components": Considering (A → B, C → D), since A and B are
```

Comment:
This is false! Consider the six - cycle X-A-B-Y-C-D-(X): swap (A-B), (C-D) -> (A-D), (C-B) . Therefore you now have the two three-cycles X-A-D-(X) and Y-C-B-(Y). We discussed this example!